### ISSUE TITLE
Is the SpeechNote.Addon.amd not supposed to support FasterWhisper STT hardware accelaration?

### ISSUE THREAD

**Issue by SugarRayLua** [flathub/net.mkiol.SpeechNote.Addon.amd#14]
Thanks for the great work on the app!

I have an AMD x86_64 Linux machine and was interested in trying out the GPU acceleration to Whisper from this add on. However, I had read that the FasterWhisper model might be even better than Whisper, so downloaded the FasterWhisper V3 model in addition to the Whisper V3 model expecting to be able to GPU accelerate both. However, on the settings menu, the option to hardware accelerate the FasterWhisper model was grayed out while the option to hardware accelerate the Whisper model was not. I was wondering if I didn't download something or do something right as to the reason I couldn't choose the option to hardware accelerate FasterWhisper or if the option for hardware acceleration for AMD FasterWhisper is just not supported yet.

Thank you!

**Comment by mkiol** [comment]
Hi. Thanks for the question.

FasterWhisper currently only supports CUDA acceleration because the core library [CTranslate2](https://github.com/OpenNMT/CTranslate2), which FasterWhisper uses, only works with CUDA.

There is an [unofficial fork of CT2 for ROCm](https://github.com/arlo-phoenix/CTranslate2-rocm/blob/rocm/README_ROCM.md) and some people reported that it works with FasterWhisper. I tried to test this, but most likely my AMD card is not compatible and the result was disappointing.

In my opinion the best option for AMD is to use WhisperCpp with Vulkan acceleration. Vulkan is supported in pre-official Speech Note 4.7.0-beta version. To use it you don't need to install any add-on. [At least one Speech Note's user with AMD card confirmed that Vulkan works wonderfully on 4.7.0-beta](https://github.com/mkiol/dsnote/issues/137). Here is more info about [Speech Note 4.7.0-beta](https://github.com/mkiol/dsnote/issues/179). 

BTW, The best place to report problems, ask question is [dsnote](https://github.com/mkiol/dsnote/issues) project. This repository is only for Flatpak package.

**Comment by SugarRayLua** [comment]
Thanks, @mkiol for the prompt and helpful reply.

I'm currently using version 4.6 from flathub with WhisperCpp and hardware with the OpenVino hardware acceleration but now notice that is actually using the CPU (AMD Ryzen 7, 8840). Would you then suggest I try downloading the 4.70 beta and using that instead with GPU acceleration?

**Comment by mkiol** [comment]
Yes, I strongly recommend trying 4.7 Beta from the `flathub-beta` channel. It is a "beta", but it is completely usable and rather stable. Yesterday I released a new 4.7 Beta 2, which fixes several Vulkan-related bugs.

After installing 4.7 Beta, set Vulkan acceleration in WhisperCpp's options. After restarting, you should see a huge performance boost with WhisperCpp models.

![image](https://github.com/user-attachments/assets/6e4deeab-f460-4843-b99d-2e3e37e1620b)

More info about Beta is [here](https://github.com/mkiol/dsnote/issues/179).

**Comment by SugarRayLua** [comment]
Thank you very much, @mkiol !

One clarification, when I actually go to install the beta from the command line, I get three options for speechnote beta and don't know if I'm supposed to install 1) or 3):

1) runtime/net.mkiol.SpeechNote.Addon.amd/x86_64/beta
2) runtime/net.mkiol.SpeechNote.Addon.nvidia/x86_64/beta
3) app/net.mkiol.SpeechNote/x86_64/beta

**Comment by mkiol** [comment]
The 3 is Speech Note app. The 1 is an add-on which gives you ROCm acceleration in the app. Since you will use only Vulkan you can skip 1.

**Comment by SugarRayLua** [comment]
Thank you, @mkiol, for your continued assistance and patience in helping me. 

I installed the beta as you said, #3, and am using the WhisperCpp-LargeV3 model but for some reason can't get Vulkan to come up as an option. Is that option only for NIVIDIA as show in your example? I'm on AMD, Ryzen with its integrated GPU and not NIVIDIA if that is the problem.

**Comment by mkiol** [comment]
It is integrated GPU? Ohh, that explains why ROCm didn't work and why you don't see Vulkan device. 

In Speech Note, by default, Vulkan is disabled for integrated GPU because most likely it will not work well. To try it, you need enable _Vulkan iGPU_ is advanced setting (Settings->Other->Show advanced settings->Hardware acceleration options->Vulkan iGPU).

![image](https://github.com/user-attachments/assets/82b31f67-21fb-4975-a9d9-0a5b62906ca4)

After restarting the app, you should see Vulkan device in WhisperCpp options.

**Comment by SugarRayLua** [comment]
Thanks again for your patience and promptness, @mkiol -- your last instructions worked!
Good speed now translating speech to text in real time :-)

Any suggestions for the future how to work with it on improving accuracy? I recently transitioned off of mobile platforms (iPAD OS for a long time and some Android) to Linux. On iPAD OS, I was using two different on device (iPad Pro) Whisper base apps:

WhisperBoard 
and
SuperWhisper

that seemed a bit more accurate in translating my sample dictations. (Have not done head to head comparisons)

**Comment by mkiol** [comment]
> Thanks again for your patience and promptness, @mkiol -- your last instructions worked!
> Good speed now translating speech to text in real time :-)

Wow. I wasn't expecting that. Especially not with an iGPU. Perhaps Vulkan support on iGPU should be enabled by default 🤔 . Can you estimate more/less how much faster decoding is with GPU acceleration enabled?

> Any suggestions for the future how to work with it on improving accuracy?

What model do you use? Large-v3 maybe? To improve accuracy you may chance _Profile_ to _Best quality_.

![image](https://github.com/user-attachments/assets/f3981712-01cb-434f-8dd7-6bd5461ecead)

**Comment by SugarRayLua** [comment]
Hi, @mkiol,

Yes, changing the Profile to Best quality-- made a lot of difference-- much better! 

Since I am in the medical profession, I test out medical dictation with speech dictation. Here is a trial run when I used my Ryzen with its integrated iGPU WhisperCpp  V3 large + Vulkan hardware acceleration with Profile now set to Best Quality:

(not a real patient-- hypothetical patient below-- actually 100% accurate for the run below):
"C.C., follow-up hypertension. H.P.I., patient is a 53-year-old man with a history of hypertension and returns for follow-up. The patient states he is doing much better on his hypertension medication. The patient states he is having no side effects and his blood pressures are now at 130s over 50s.
Review of systems, notable that the patient recently had an upper respiratory tract infection. Physical exam, vitals, temp 98, pulse 80, respiration 16, blood pressure 130 over 70. General, the patient is well-appearing and no acute distress. Heart, rate, rhythm, rhythm, without murmurs or gallops, no edema. Labs, notable for a normal creatinine. Assessment plan, number one, hypertension. Is much better controlled now. Will continue his current antihypertensive medications and follow up in one month."

Perhaps in the future on your website you could put together a suggested list of settings for different use cases?

Perhaps you might wish in the future to add in a small on device AI that could be used to format dictations (e.g. set up the above dictation into an actual note format for a medical record? That is what I see some of the other Whisper dictation apps doing [although for privacy would prefer to do so with an AI on device])

I don't have a great feel for how "much more" the Vulkan acceleration helped for iGPU setting, but it was definitely noticeable. If you'd like, when I'm off work, I'd be happy to try and send a video/audio recording of hypothetical dictation like the one above for you to judge for yourself how well Vulkan works for iGPU. :-)

**Comment by mkiol** [comment]
> Yes, changing the Profile to Best quality-- made a lot of difference-- much better!

I'm glad it helped!

> Since I am in the medical profession, I test out medical dictation with speech dictation. 

I just did a quick search on [HuggingFace website](https://huggingface.co/models?search=whisper%20medical) and it looks like there are plenty of "fine-tuned" Whisper models for medical vocabulary. Perhaps I could add one of two of them into Speech Note 🤔 

> you might wish in the future to add in a small on device AI that could be used to format dictations (e.g. set up the above dictation into an actual note format for a medical record?

This is very interesting, but also a very specific use case. Speech Note is a general purpose note taking app, so this type of feature may not be the best fit.

> If you'd like, when I'm off work, I'd be happy to try and send a video/audio recording of hypothetical dictation like the one above for you to judge for yourself how well Vulkan works for iGPU.

That's very kind of you. It won't be necessary. I think I know what I need to know. I am glad that you are testing the Beta version. If you notice any problem, please report it.
